sklearn/utils/_metadata_requests.py

```diff
@@ -1251,6 +1254,56 @@ class _MetadataRequester:
     .. versionadded:: 1.3
     """
 
+    if TYPE_CHECKING:
+        # This code is never run in runtime, but it's here for type checking.
+        # Type checkers fail to understand that the `set_{method}_request`
+        # methods are dynamically generated, and they complain that they are
+        # not defined. We define them here to make type checkers happy.
+        # During type checking analyzers assume this to be True.
+        # The following list of defined methods mirrors the list of methods
+        # in this list:
+        # SIMPLE_METHODS = [
+        #     "fit",
+        #     "partial_fit",
+        #     "predict",
+        #     "predict_proba",
+        #     "predict_log_proba",
+        #     "decision_function",
+        #     "score",
+        #     "split",
+        #     "transform",
+        #     "inverse_transform",
+        # ]
```

Comment:
Given this is for type checking, I think we can reference the `SIMPLE_METHODS` variable and look for it in the file.

```suggestion
        # in `SIMPLE_METHODS`.
```

This way we do not need to keep them in sync if a new method is added to `SIMPLE_METHODS`.